src/ipc/ipc4/helper.c

```diff
@@ -101,7 +104,7 @@ struct comp_dev *comp_new(struct sof_ipc_comp *comp)
 		const struct ipc_config_process spec = {
 			.data = (const unsigned char *)MAILBOX_HOSTBOX_BASE,
 			/* spec_size in IPC4 is in DW. Convert to bytes. */
-			.size = comp->ext_data_length * 4,
+			.size = module_init->extension.r.param_block_size * sizeof(uint32_t),
```

Comment:
So this is indeed easier to follow, FYI @ujfalusi , here we get the payload size to use in module code (for parsing module specific data).